Java/core/src/test/java/com/amazon/randomcutforest/CPUTest.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazon.randomcutforest;
+
+import com.amazon.randomcutforest.testutils.ShingledMultiDimDataWithKeys;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+
+import java.util.Arrays;
+import java.util.concurrent.ForkJoinPool;
+
+/**
+ * The following "test" is intended to provide an approximate estimate of the improvement
+ * from parallelization. At the outset, we remark that running the test from inside
+ * an IDE/environment may reflect more of the environment. Issues such as warming are not
+ * reflected in this test.
+ *
+ * Users who wish to obtain more calibrated estimates should use a benchmark -- preferably
+ * using their own "typical" data and their end to end setup. Performance of RCF is data dependent.
+ * Such users may be invoking RCF functions differently from a standard "impute, score, update"
+ * process recommended for streaming time series data.
+ *
+ * Moreover, in the context of a large number of models, the rate at which the models require
+ * updates is also a factor and not controlled herein.
+ *
+ * The two tests should produce near identical sum of scores, and (root) mean squared error of
+ * the impute up to machine precision (since the order of the arithmetic operations would vary).
+ *
+ * To summarize the lessons, it appears that parallelism almost always helps (upto resource limitations).
+ * If an user is considering a single model -- say from a console or dashboard, they should consider
+ * having parallel threads enabled. For large number of models, it may be worthwhile
+ * to also investigate different ways of achieving parallelism and not just attempt to
+ * change the executor framework.
+ *
+ */
+
+@Tag("functional")
+public class CPUTest {
```

Comment:
What's the rationale for including this here and not in the benchmark package?

Reply:
Resourcing. A good benchmarking needs calibration and needs to handle all of the caveats mentioned in the comments. This simple test helps reveal to an user which path they need to follow, and should be of interest because enabling parallelism (toggling the numberOfThreads) shows improvement in both async/sync versions, contrary to experiments in some prior version.